std/assembly/date.ts

```diff
@@ -337,13 +310,13 @@ function ymdFromEpochDays(z: i32): i32 {
 }
 
 // http://howardhinnant.github.io/date_algorithms.html#days_from_civil
-function daysSinceEpoch(y: i32, m: i32, d: i32): i32 {
+function daysSinceEpoch(y: i32, m: i32, d: i32): i64 {
   y -= i32(m <= 2);
   var era = <u32>floorDiv(y, 400);
   var yoe = <u32>y - era * 400; // [0, 399]
   var doy = <u32>(153 * (m + (m > 2 ? -3 : 9)) + 2) / 5 + d - 1; // [0, 365]
   var doe = yoe * 365 + yoe / 4 - yoe / 100 + doy; // [0, 146096]
-  return era * 146097 + doe - 719468;
+  return <i64><i32>(era * 146097 + doe - 719468);
```

Comment:
Can be shortened to
```suggestion
  return <i32>(era * 146097 + doe - 719468);
```
but if you prefer the explicit cast for clarity, I am also OK with keeping it.

Reply:
I prefer more explicit cast here